src/persist-client/src/internal/state.rs

```diff
@@ -851,17 +850,13 @@ pub struct State<K, V, T, D> {
     pub(crate) hostname: String,
     pub(crate) collections: StateCollections<T>,
 
-    // According to the docs, PhantomData is to "mark things that act like they
-    // own a T". State doesn't actually own K, V, or D, just the ability to
-    // produce them. Using the `fn() -> T` pattern gets us the same variance as
-    // T [1], but also allows State to correctly derive Send+Sync.
-    //
-    // [1]:
-    //     https://doc.rust-lang.org/nomicon/phantom-data.html#table-of-phantomdata-patterns
-    pub(crate) _phantom: PhantomData<fn() -> (K, V, D)>,
+    pub(crate) key_codec: String,
```

Comment:
thinking of collapsing all these into a single `codecs: (String, String, String, String)`